- add Room database

dencharski opened the pull request.

dencharski committed:
- add Room database

i-masloed commented on .idea/kotlinc.xml:
```diff
@@ -1,6 +1,6 @@
 <project version="4">
   <component name="KotlinJpsPluginSettings">
-    <option name="version" value="1.8.20" />
+    <option name="version" value="1.9.0" />
```
Рекомендация: файлы директорий .idea и build стоит вынести в .gitignore

i-masloed commented on app/src/main/java/com/example/playlistmaker/TrackDtoApp.kt:
```diff
@@ -2,6 +2,7 @@ package com.example.playlistmaker
 
 import android.os.Parcel
 import android.os.Parcelable
+import android.util.Log
```
Этот класс - модель данных. В соответствии с правилом зависимостей должн лежать в слое domain

i-masloed commented on app/src/main/java/com/example/playlistmaker/TrackDtoApp.kt:
```diff
@@ -33,14 +35,20 @@ data class TrackDtoApp(
         parcel.readString().toString(),
         parcel.readString().toString(),
-        parcel.readString().toString()
+        parcel.readString().toString(),
```
этот класс Parcelale и использует плагин Parcelize, поэтому не нужно писать самостоятельно  "распаковку"  и запись объекта - плагин сам сгененирует нужный код при компиляции. Все , что нужно - это добавить интерфейс в сигнатуру (`: Parcelable`) и добавить аннотацию @Parcelize

i-masloed commented on app/src/main/java/com/example/playlistmaker/search/ui/SearchFragment.kt:
```diff
@@ -227,7 +227,7 @@ class SearchFragment : Fragment(), TrackListAdapter.ItemClickInterface,
 
         intent.putExtra(
-            App.trackKey, TrackDto(
+            App.trackKey, TrackDtoAudioPlayer(
```
а почему бы здесь просто не передать track? он же Parcelable
`intent.putExtra(App.trackKey, track)`

i-masloed commented on app/src/main/java/com/example/playlistmaker/mediateka/ui/SelectedTracksViewModel.kt:
```diff
@@ -1,6 +1,37 @@
+class SelectedTracksViewModel(private val selectedTrackInteractor: SelectedTrackInteractor) :
+    ViewModel() {
+    private val _selectedTracksViewState = MutableLiveData<SelectedTracksViewState>()
+    val selectedTracksViewState get() = _selectedTracksViewState
```
не прописан тип возвращаемой переменной, соответственно она автоматом приводится к MutableLiveData

i-masloed commented on app/src/main/java/com/example/playlistmaker/mediateka/ui/SelectedTracksFragment.kt:
```diff
@@ -28,15 +44,101 @@ class SelectedTracksFragment : Fragment() {
+        )
+
+        intent.putExtra(
+            App.trackKey, TrackDtoAudioPlayer(
```
почему бы просто не передать track?

dencharski committed:
- add Room database.
fixed the display of the track status and the "like" button.
TrackDtoApp - ui class, removed from the date layer.
